### PR TITLE
[gateway-internal] add new internal gateway

### PR DESCRIPTION
**Pull request by jigold**
@cseed The part I am stuck on is the authentication for the router resolver. How does the batch2 instance in a test namespace get access to the real encryption key that the router resolver is expecting? Can you also double check the nginx configuration?

### CODE REVIEW

File: gateway-internal/get-deployed-sha.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (danking) [inline gateway-internal/get-deployed-sha.sh]: I don't think these files are required any more.

File: gateway-internal/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    matchLabels:
+      app: gateway-internal
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: gateway-internal
+        hail.is/sha: "{{ code.sha }}"
+    spec:
+      serviceAccountName: gateway-internal
```
Comment (danking) [inline gateway-internal/deployment.yaml]: Does this service account exist?

**Comment by danking** [comment]
It might be easier to invent a new domain name for VPC-internal clients. `batch.hail.internal`. Add the right DNS routes. Add new root certs to the worker nodes. That ensures our DNS changes don't break anyone else who is trying to talk to `batch.hail.is`. We should maybe talk more about what these DNS changes are breaking so we can plan our network infrastructure.

File: gateway-internal/deployment.yaml (hunk 2 above)
Comment (jigold) [reply gateway-internal/deployment.yaml]: No. I'm not sure where the gateway service account got defined but it does exist...

Comment (jigold) [reply gateway-internal/deployment.yaml]: It looks like the gateway service account has a secret in its configuration. Can we discuss what the role of the service accounts are so I can make the proper one (or delete) for the internal gateway?

File: hail/python/hailtop/gear/auth/hailjwt.py [hunk 3]
```diff
@@ -107,20 +108,20 @@ def parse_header(request):
...
-    token_getter = lambda request: parse_header(request)
-    return authenticated_users_only(token_getter)(fun)
+    return authenticated_users_only(parse_header)(fun)
 
 
 def rest_authenticated_developers_only(fun):
-    token_getter = lambda request: parse_header(request)
-    return authenticated_developers_only(token_getter)(fun)
+    return authenticated_developers_only(parse_header)(fun)
+
```
Comment (jigold) [inline hail/python/hailtop/gear/auth/hailjwt.py]: I can't figure out why this file and the one below still show up as changes. I checked them out from master to try and revert my changes...

**Comment by jigold** [comment]
Something still isn't right with my configuration. The service and deployment are all up. I can curl to the internal gateway and it shows up in the logs. However, I'm getting 404 with this query: `http://hail.internal/jigold/batch2/healthcheck`. I set up cloud dns to route hail.internal to the internal gateway. I verified no traffic is recorded in the jigold router. I tried adding and removing the `gateway` service account but it made no difference.

**Pull request merged**